lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -111,26 +126,54 @@ int event_handler_list_remove_handler(lte_lc_evt_handler_t handler)
 /**@brief dispatch events. */
 void event_handler_list_dispatch(const struct lte_lc_evt *const evt)
 {
-	struct event_handler *curr, *tmp;
+	struct lte_lc_event_fifo_item *lte_lc_event_item;
 
 	if (event_handler_list_is_empty()) {
 		return;
 	}
 
+	lte_lc_event_item = k_heap_alloc(
+				&lte_lc_event_heap,
+				sizeof(struct lte_lc_event_fifo_item),
+				K_NO_WAIT);
+	if (!lte_lc_event_item) {
+		LOG_WRN("No heap space for outgoing lte_lc event type=%d", evt->type);
+		__ASSERT(
+			lte_lc_event_item,
+			"No heap space for lte_lc notification type=%d",
+			evt->type);
+		return;
+	}
+
+	memcpy(&lte_lc_event_item->evt, evt, sizeof(struct lte_lc_evt));
```

Comment:
Yep, need to see how this can be resolved. Need to have some additional dynamic memory allocation for those to avoid always reserving the memory for NCELLMEAS results.

Reply:
I've made now allocation and free functions for the event structures. At this point they are using k_malloc and k_free for ncells and gci_cells but that can be changed once we decide where we want to allocate them.